### DISCUSSION TITLE
Transfer mailboxes

### DISCUSSION THREAD

**Discussion by karlkashofer** [docker-mailserver/docker-mailserver#3738] [Q&A]
Hi ! 
I run a postfix server with virtual mailboxes (virtual_mailbox_base = /home/vmail) and would like to move to DMS.
I have set up DMS and it seems to work, but now i need to transfer all users and their emails from my old postfix to my new.

How is that done ? 
Can i just create the corresponding users on DMS and copy all mail from the old vmail "cur" folder into the users mail data folder ?
If yes, how is that affecting imap folder structure ?

**Comment by casperklein** [comment]
Never used myself, but you may take a look at https://github.com/imapsync/imapsync.

**Comment by MarcS1975** [comment]
I have used it and it is fantastic.
All you need is the login details for both OLD and NEW servers for each user and imapsync transfers everything (folders, emails etc).
I even managed to transfer MS Exchange accounts into my DMS.

You might have to tweak some parameters but you can do dry-runs and imapsync gives extensive log-files so you can see every step. In the log files Imapsync suggests to tweak certain parameters based on the encountered errors.
I had to amend my /etc/ssl/openssl.cnf slightly (which is only necessary for the transfer period)
There is a lot of [Q&A / FAQs available online](https://imapsync.lamiral.info/FAQ.d/) so any error messages you see in the logs will most likely find a solution.
